plugins/module_utils/models.py

```diff
@@ -82,17 +82,32 @@ def from_json(cls, json: TerraformModuleResource) -> "TerraformAnsibleProvider":
 @dataclass
 class TerraformChildModule:
     resources: List[TerraformModuleResource]
+    child_modules: List["TerraformChildModule"] = field(default_factory=lambda: [])
+
+    def get_resources(self) -> List[TerraformModuleResource]:
+        return self.resources + sum([child.get_resources() for child in self.child_modules], [])
 
     @classmethod
     def from_json(cls, json: TJsonObject) -> "TerraformChildModule":
-        return cls(resources=[TerraformChildModuleResource.from_json(r) for r in json.get("resources", [])])
+        return cls(
+            resources=[TerraformChildModuleResource.from_json(r) for r in json.get("resources", [])],
+            child_modules=[TerraformChildModule.from_json(r) for r in json.get("child_modules", [])],
+        )
 
 
 @dataclass
 class TerraformRootModule:
     resources: List[TerraformModuleResource]
     child_modules: List[TerraformChildModule]
 
+    def get_resources(self, search_child_modules: bool) -> List[TerraformModuleResource]:
+        child_resources: List[TerraformModuleResource] = []
+
+        if search_child_modules:
+            child_resources = sum([child.get_resources() for child in self.child_modules], [])
+
+        return self.resources + child_resources
+
```

Comment:
This dilutes the data model since `TerraformRootModule.resources` is exactly the same thing as `TerraformRootModule.get_resources(False)`. `get_resources()` is a bit misleading since `TerraformRootModule.resources` already exist. It would be better call this function something like `flatten_resources()` and have it not accept any arguments. The same thing goes for `TerraformChildModule`.